### DISCUSSION TITLE
Tile available for specific LatLon (utility function?)

### DISCUSSION THREAD

**Discussion by videgro** [mapsforge/mapsforge#1240] [Q&A] [answered]
Dear all,

Maybe I have over looked it, but is there a function/utility to find whether or not a specific LatLon (position) is available at the currently selected MapDataStore?

I know there is a method to get the BoundingBox of a map. This is a rectangle. But (some) maps do not have tiles for the complete rectangle.

By the way, I'm developing for Android.

Thanks for your answer in advance!

**Comment by devemux86** [comment, marked as answer]
The maps are rectangular and are defined by their bounding box.

If you know the region then you can check the .poly file from [Geofabrik](https://download.geofabrik.de/).

**Reply by videgro** [reply]
Thanks for your reply, but I think I still don't understand something of MapsForge.

I downloaded the map of The Netherlands from https://download.mapsforge.org/maps/v5/europe/netherlands.map
Which has the BoundingBox: minLatitude=50.64753, minLongitude=2.892192, maxLatitude=54.11786, maxLongitude=7.330455

Now I have a LatLong position in Germany: latitude=50.6792, longitude=7.248638333333333

When I request: boundingbox.contains(position) this is TRUE. Yes indeed, the position is within the limits of the BoundingBox. But when I render the map, I don't get tiles. I just get the gray background.

**Reply by devemux86** [reply]
The maps in Mapsforge server are created with data from [Geofabrik](https://download.geofabrik.de/).

See the [Geofabrik Netherlands](https://download.geofabrik.de/europe/netherlands.html) region.
Its map shows the polygon filled with data.

**Reply by videgro** [reply]
It's true at the polygon at Geofabrik for the Netherlands,  Königswinter (Germany, NRW) is not included. So I would like to motivate the user to download a map where there is data (tiles) available for that region. But how can I detect that in the map of the Netherlands, there is no data for Königswinter available, but the the boundingbox of the Netherlands, contains Königswinter?

**Reply by devemux86** [reply]
The polygonal bounds with filled data for each country / region exist in the [Geofabrik](https://download.geofabrik.de/) polygons.
You can download the .poly files and read / use them with [JTS](https://github.com/locationtech/jts) library to check if they contain a location.

Or you can create your own maps with filled the entire rectangle.
This is a different process with input the planet.osm (for details see the old [mapsforge-mapcreator](https://github.com/mapsforge/mapsforge-mapcreator)).

**Reply by devemux86** [reply]
Or use maps from another provider, like [OpenAndroMaps](https://www.openandromaps.org/en).
